lib/connection/connection.go

```diff
@@ -13,6 +13,9 @@ func GetIrcConn(cfg *configure.Config) (*irc.Connection, error) {
 	conn.UseTLS = cfg.Server.UseTls
 	conn.VerboseCallbackHandler = cfg.Connection.VerboseCallbackHandler
 	conn.Debug = cfg.Connection.Debug
+	if cfg.Server.ServerAuth {
```

Comment:
The `ServerAuth` config option is redundant. You can just check if `ServerPassword` is non-empty. This is how nickserv identify works.